accounting/src/delta/delta_amount_collection.rs

```diff
@@ -58,28 +61,29 @@ impl<K: Ord> DeltaAmountCollection<K> {
         key: K,
         other_amount: SignedAmount,
     ) -> Result<(), Error> {
-        let current = self.data.get(&key);
-        let new_bal = Self::combine_signed_amount_delta(&current.copied(), other_amount)?;
-        if new_bal == SignedAmount::ZERO {
-            // if the new amount is zero, no need to have it at all since it has no effect
-            self.data.remove(&key);
-        } else {
-            self.data.insert(key, new_bal);
+        match self.data.entry(key) {
+            Entry::Occupied(mut current) => {
+                let new_bal = Self::combine_signed_amount_delta(*current.get(), other_amount)?;
+                // if the new amount is zero, no need to have it at all since it has no effect
+                if new_bal == SignedAmount::ZERO {
+                    drop(current.remove_entry());
+                } else {
+                    *current.get_mut() = new_bal;
+                }
+            }
+            Entry::Vacant(entry) => {
+                entry.insert(other_amount);
+            }
```

Comment:
This seems to me more unnecessarily verbose.

If you'd like to get rid of that `Option<SignedAmount>` in `combine_signed_amount_delta`, just change `current.get()` to `*current.unwrap_or(&SignedAmount::ZERO)`... not a huge benefit though.

Reply:
It was more about replacing the get + remove/insert on the map than the Option but ok.

Reply:
reverted